src/common/ZDMarker.ts

```diff
@@ -69,11 +69,7 @@ export class ZDMarker extends Marker {
         marker.complete();
       });
       marker.popup.on("uncomplete", () => {
-        const tag = marker.tags.indexOf("Completed");
-        if (tag > -1) {
-          marker.tags.splice(tag, 1);
-        }
-        marker.fire("uncompleted");
+        marker.clearCompletion();
```

Comment:
causes a redundant call to popup.markUncompleted but it's harmless

Reply:
> causes a redundant call to popup.markUncompleted but it's harmless

This matches the redundant call to popup.markCompleted that's already present in the code!